openmc/mesh.py

```diff
@@ -570,19 +595,19 @@ def __repr__(self):
         fmt = '{0: <16}{1}{2}\n'
         string = super().__repr__()
         string += fmt.format('\tDimensions', '=\t', self.n_dimension)
-        x_grid_str = str(self._x_grid) if not self._x_grid else len(self._x_grid)
+        x_grid_str = str(self._x_grid) if self._x_grid is None else len(self._x_grid)
```

Comment:
Why make these changes to use `is None`, `is not None`, etc? It's a very common python idiom to say things like `if self._y_grid` rather than `self._y_grid is not None`.

Feel free to mark this as resolved if anyone thinks it's absolutely necessary that we explicitly compare to None.

Reply:
I didn't make this change, but I believe this update ensures the correct behavior when a grid is empty, but not `None`. For example, an empty Python list would evaluate as `False` here and result  `"[]"` instead of the length of the list.